packages/atomic/.storybook/preview-head.html

```diff
@@ -1,3 +1,13 @@
 <script type="module" src="atomic/atomic.esm.js"></script>
 <script nomodule src="atomic/atomic.js"></script>
+<script>
+  window.addEventListener('load', () => {
+    setTimeout(() => {
+      let message = {
+        height: document.body.scrollHeight,
+      };
+      window.top.postMessage(message, '*');
```

Comment:
Hmm, is a CSS solution not possible? I believe setting height and width to 100% on an iframe element will prevent scrollbars,

```
<iframe style="height: 100%; width: 100%;">
```

Reply:
@samisayegh  unfortunately, it doesn't work. This is what I tried first :D 

I tried playing with CSS as long as possible because I did want to make the solution simpler. Eventually, it's never "simpler" with iframes, lol.

Reply:
Usually, `height: 100%` works only if you have the same height value in the parent container and its parent's container, and so on up to the `html` tag itself. I used it back in the day as a well-known trick for creating screens that fill the whole viewport but don't create vertical scrollbars at the same time (it was useful for creating first screens on landing pages, for example). 

In my case, I don't want (and not sure if smb would allow me to) change the docs styles SO drastically :) This is why I've been digging into how to tame the iframe itself…

Reply:
Ok, I see where we are differing!

Yes, with those styles, you would need to set a fixed placeholder height on the parent element (I agree, inheriting 100% to the root is not desirable).

Then, we javascript, you could monitor the height of the iframe, and adjust the height of the parent to ensure no scrollbars?

Could this approach avoid needing to post message across frames?

Reply:
## Short answer:
Unfortunately, we **cannot** avoid sending messages between windows.
The only way for cross-domain interactions is using the `postMessage` method in the frame and listening to the incoming messages in the parent window.

---

## Long answer

### iframe window (storybook story)

Each story' `body` has a different height that is defined by the size of content.

### Parent window (doc page)

The styles define that every `iframe` block should be strictly 150 px in height. Doesn't matter how big/small iframe content is.
This is why I see the vertical scrollbar in the frame.

### Interaction between windows

When the parent window is loading, I do not want the iframe height to be 150 px, I want it to be updated according to the content size.

The approach is straightforward: the parent page should get the height of the `iframe's body` and then assign this value to the `iframe.style.height`. In this way, an iframe in the parent window would be as high as its actual content.

BUT! Due to the security policies, JS in the parent window CANNOT directly get anything from the iframe. This is because our windows has different origins. Thus, I cannot get the `iframe's body` by running a script in the parent window.

The only way for cross-domain interactions is using the `postMessage` method in the frame and listening to the incoming messages in the parent window.

In the parent window, I'm using the following code (don't pay attention to the link in the iframe, it's just a temporary local dev link):

```html
<iframe frameborder="0" width="100%" id="componentIframe" src="http://localhost:6006/iframe.html?id=atomic-categoryfacet--default-category-facet&args=&viewMode=story"></iframe>

<script>

  let iframe = document.querySelector("#componentIframe");

  window.addEventListener('message', (e) => {
   
    let message = e.data;
    iframe.style.height = message.height + 10 + 'px';

  } , false);

</script>
```